src/Component/Context/Option/MetadataOption.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Context\Option;
+
+use Sylius\Component\Resource\Metadata\MetadataInterface;
+
+final class MetadataOption
+{
+    public function __construct(private MetadataInterface $metadata)
+    {
+    }
+
+    public function metadata(): MetadataInterface
+    {
+        return $this->metadata;
```

Comment:
Why did you introduce these wrapping objects?

Reply:
I've added a comment on the PR description itself.